scripts/pylib/twister/twisterlib/environment.py

```diff
@@ -363,8 +364,11 @@ def parse_arguments(args):
                         help="Specify a file where to save logs.")
 
     parser.add_argument(
-        "-M", "--runtime-artifact-cleanup", action="store_true",
-        help="Delete artifacts of passing tests.")
+        "-M", "--runtime-artifact-cleanup", choices=['passing-tests', 'all-tests'],
+        default=None, const='passing-tests', nargs='?',
+        help="""This option allows to delete artifacts of tests. The default value
```

Comment:
`Cleanup test artifacts; The default behavior is 'pass' which only removed artifacts of passing tests, if you wish to remove all artificats including those of failed tests, use 'all'`